### PR TITLE
numpy 2.0 compatability

### LINKED ISSUE #1186: numpy 2.0
see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#numpy-2-migration-guide

The migration depends on upstream packages--namely tidy_headers.  In the meantime, we can prevent issues by pinning our install to numpy<2.0.

### PR DESCRIPTION
**Pull request by ddkohler**
Making the transition to support the new API
https://numpy.org/doc/stable/numpy_2_0_migration_guide.html

## Changes

* numpy v2 compliant
* resolves #1186
* pin sphinx<8.0
* addresses #1192 

## Checklist

- [x] updated documentation (update "from" docs, example)
- [x] updated CHANGELOG.md
- [x] tests pass

### CODE REVIEW

**Comment by ddkohler** [comment]
everything looks ready on our end; just waiting on an upstream fix (https://github.com/untzag/tidy_headers/pull/20) to get this through

**Comment by untzag** [comment]
Hi @ddkohler @rpm4, tidy_headers has finally been released to both PyPI and conda-forge

File: setup.py
Comment (kameyer226) [inline setup.py:1]: sphinx<8.0 ?

Comment (ddkohler) [reply setup.py:1]: @kameyer226 see #1192

**Review by kameyer226 — APPROVED** [review]

**Pull request merged**